### PR TITLE
Update handlebars minimum version

### PR DESCRIPTION
**Pull request by alecharp**
Replaces #5372.

Diff: https://github.com/handlebars-lang/handlebars.js/compare/v3.0.3...v3.0.8

Note: based on `yarn.lock`, the version used in recent build was already `3.0.8`.

### Proposed changelog entries

* Entry 1: Remove unnecessary bundled handlebars asset 
* ...

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [x] (If applicable) Jira issue is well described
- [x] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [x] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@fqueiruga @jenkinsci/core-pr-reviewers 

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

**Comment by timja** [comment]
> Update Handlebars version to 3.0.8

But you're not? It's already using 3.0.8

Aren't you removing the old bundled handlebars? is there any impact to that (I suspect not)

**Comment by alecharp** [comment]
> But you're not? It's already using 3.0.8

You are correct. 

The impact could be for plugins using the asset without declaring the dependency. For the setup wizard, it's already using the NPM one thanks to the webpack build process.

**Review by timja — APPROVED** [review]

**Review by jtnord — APPROVED** [review]

**Review by res0nance — APPROVED** [review]

**Review by rsandell — APPROVED** [review]

**Review by StefanSpieker — APPROVED** [review]

File: test/src/test/java/jenkins/model/AssetManagerTest.java
Comment (MRamonLeon) [inline test/src/test/java/jenkins/model/AssetManagerTest.java:52]: Is there any change in the way the plugins consume handlebars3 now? Do plugin developers need to do any change? or are the plugins going to stop working because of that or trigger a 404 in the browser console?

Comment (alecharp) [reply test/src/test/java/jenkins/model/AssetManagerTest.java:52]: They would have to declare the dependency to Handlebars plugin or use the webpack process that was done on Jenkins War and few plugins.

**Comment by oleg-nenashev** [comment]
Do we have a list of the plugins which will be affected by the change and require updates?
There are multiple references in plugins (https://github.com/search?l=JavaScript&p=1&q=org%3Ajenkinsci+handlebars3&type=Code) , but I am not sure whether the impact was evaluated

**Comment by alecharp** [comment]
I don't have a list. From this search, we can see references from `gulpfile.js` with a dependency to the handlebars plugin in the `pom.xml`, references in `target/` folder (the plugin would need a cleanup). 

Initially, only `pipeline-stage-view` plugin was listed in http://stats.jenkins.io/jenkins-stats/jenkinsgraph.html?filter=handlebars as having a dependency to `handlebars` plugin. I don't know why `codehealth` plugin was not listed.

**Comment by MRamonLeon** [comment]
retriggering CI to see whether telemetry test failures are flakies (likely)

**Comment by MRamonLeon** [comment]
The test failure is not related. It's fixed in https://github.com/jenkinsci/jenkins/pull/5398

**Comment by alecharp** [comment]
I've been looking at plugins using Handlebars in JavaScript. First, there is only a few of them:
* jenkinsci/build-line-plugin bundles its own version of the library
* jenkinsci/test-results-analyzer-plugin bundles its own version of the library
* jenkinsci/xray-connector-plugin bundles its own version of the library
* jenkinsci/xray-connector-plugin bundles its own version of the library
* jenkinsci/codehealth-plugin depends on the handlebars plugin for Jenkins
* jenkinsci/build-pipeline-plugin bundles its own version of the library
*  jenkinsci/pipeline-stage-view-plugin depends on the handlebars plugin for Jenkins
* jenkinsci/uithemes-plugin has a NPM / Gulp build process
* jenkinsci/test-results-analyzer-plugin bundles its own version of the library

I do not foresee any issue removing the asset-core binary from Jenkins WAR.

**Review by MRamonLeon — APPROVED** [review]

**Comment by oleg-nenashev** [comment]
Maybe we should put it on hold for one week until the LTS baseline selection, just in case there are regressions.
No strong opinion, fine with merge as well. Taking all other reviews, will mark for merge so that we start the countdown.

We may merge it in 24 hours if there is no negative feedback. Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process

**Comment by jtnord** [comment]
> Maybe we should put it on hold for one week until the LTS baseline selection

this is something that would be great to get into the next LTS so holding of a weekly would not be so great from my PoV.

**Pull request merged**
